File: app/src/main/java/com/riders/thelab/core/service/ScheduleAlarmService.java
package com.riders.thelab.core.service;

import android.app.Service;
import android.content.Intent;
import android.os.Binder;
import android.os.IBinder;

import androidx.annotation.Nullable;

import timber.log.Timber;

public class ScheduleAlarmService extends Service {

    public static final int STATUS_RUNNING = 0;
    public static final int STATUS_FINISHED = 1;
    public static final int STATUS_ERROR = 2;

    private IBinder mBinder = new Binder();

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        Timber.d("Service bound");
        return mBinder;
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        return START_NOT_STICKY;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        Timber.d("Service created");
    }

    @Override
    public void onDestroy() {
        if (null != mBinder)
            mBinder = null;

        super.onDestroy();
    }
}

File: feature/schedule/src/main/java/com/riders/thelab/feature/schedule/core/ScheduleAlarmService.kt
package com.riders.thelab.feature.schedule.core

import android.app.Service
import android.content.Intent
import android.os.Binder
import android.os.IBinder
import timber.log.Timber

class ScheduleAlarmService : Service() {
    companion object {
        const val STATUS_RUNNING = 0
        const val STATUS_FINISHED = 1
        const val STATUS_ERROR = 2
    }

    private var mBinder: IBinder? = Binder()

    override fun onBind(intent: Intent?): IBinder? {
        Timber.d("Service bound")
        return mBinder
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        return START_NOT_STICKY
    }

    override fun onCreate() {
        super.onCreate()
        Timber.d("Service created")
    }

    override fun onDestroy() {
        if (null != mBinder) mBinder = null
        super.onDestroy()
    }
}